dll/win32/shell32/shlexec.cpp

```diff
@@ -458,6 +458,144 @@ static HRESULT SHELL_GetPathFromIDListForExecuteW(LPCITEMIDLIST pidl, LPWSTR psz
     return hr;
 }
 
+static HWND SHELL_GetUsableDialogOwner(HWND hwnd)
+{
+    // Explicitly block the shell desktop listview from becoming the owner (IContextMenu calling ShellExecute)
+    HWND progman = GetShellWindow();
+    if (hwnd && IsWindowVisible(hwnd) && hwnd != GetDesktopWindow() &&
+        hwnd != progman && !IsChild(progman, hwnd))
+    {
+        return hwnd;
+    }
+    return NULL;
+}
+
+/*************************************************************************
+ *    SHELL_PromptAndRunProcessAs [Internal]
+ */
+typedef struct {
+    WCHAR NameBuffer[MAX_PATH], Pass[MAX_PATH];
+    LPWSTR Name, Domain;
```

Comment:
```suggestion
typedef struct _RUNASDLGDATA
{
    WCHAR NameBuffer[MAX_PATH];
    WCHAR Pass[MAX_PATH];
    LPWSTR Name;
    LPWSTR Domain;
```
and please rename "Pass" to "Password" as well.